ansible/mnesia_backup.yml

```diff
@@ -16,33 +16,66 @@
 - name: Backup, archive and download Mnesia database
   hosts: "{{ ansible_play_hosts | last }}"
   remote_user: aeternity
+  tasks:
+    - name: Include configuration variables
+      include_vars: backup/config.yml
 
-  vars:
-    aeternity_bin: "{{ ansible_env.HOME }}/node/bin/aeternity"
-    download_dir: /tmp/mnesia_backups
-    backup_suffix: backup
-    backup_filename: "mnesia_{{ ansible_ssh_host|default(inventory_hostname) }}_{{ backup_suffix }}"
+    - name: Disable mining and peers
+      import_tasks: backup/disable_mining_and_peers.yml
+
+    - meta: flush_handlers
+    - debug:
+        var: health_check
+
+    - debug:
+        var: project_root
+    - name: Wait aeternity node API to boot
+      wait_for:
+        port: "{{ node_config.http.external.port }}"
+        host: "{{ public_ipv4 }}"
+        timeout: 300
+      when: health_check
+
+      tags: [health-check]
 
-  tasks:
     - name: Backup Mnesia database
-      command: "{{ aeternity_bin }} eval 'ok = mnesia:backup(\"/tmp/{{ backup_filename }}\").'"
+      command: "{{ aeternity_bin }} eval 'ok = mnesia:backup(\"/tmp/backup\").'"
       args:
-        creates: "/tmp/{{ backup_filename }}"
+        creates: "/tmp/backup"
+      notify: "stop aeternity daemon"
 
     - name: Archive Mnesia backup
       archive:
-        path: /tmp/{{ backup_filename }}
-        dest: /tmp/{{ backup_filename }}.gz
+        format: zip
+        path: /tmp/backup
+        dest: /tmp/{{ backup_archive_filename }}
+
+    - name: Upload backup to S3 bucket
+      aws_s3:
+        bucket: "{{ backup_bucket }}"
+        object: "{{ backup_archive_filename }}"
+        src: /tmp/{{ backup_archive_filename }}
+        mode: put
 
-    - name: Download the Mnesia backup archive
-      fetch:
-        src: /tmp/{{ backup_filename }}.gz
-        dest: "{{ download_dir }}"
+    - name: Upload backup to S3 bucket latest
```

Comment:
Isn't it more effective to do a s3 cp of the above object ? Both in term of speed and cost ?

Reply:
hm... not sure will aws cp will be smart enought to copy it without downloading, worth to try.

Reply:
```get
put
delete
create
geturl
getstr
delobj
list```


module do not have CP option